lib/identity_cache/query_api.rb

```diff
@@ -106,6 +106,22 @@ def prefetch_associations(associations, records)
         nil
       end
 
+      protected
+
+      def record_from_coder(coder) #:nodoc:
+        if coder
+          record = instantiate(coder[:attributes].dup)
+
+          coder[:associations].each do |name, value|
+            associated_class = reflect_on_association(name).klass
+            set_embedded_association(record, name, hydrate_association_target(associated_class, value))
+          end if coder.has_key?(:associations)
```

Comment:
style nit: I'd go for more nesting

```

if coder.has_key?(...)
  loop here
end

```

take it or leave it

Reply:
I hadn't touched this line, other than making the method protected, but sure.